libcudacxx/include/cuda/std/detail/libcxx/include/complex

```diff
@@ -792,7 +1021,101 @@ operator-(const _Tp& __x, const complex<_Tp>& __y)
 }
 
 template<class _Tp>
-inline _LIBCUDACXX_INLINE_VISIBILITY _LIBCUDACXX_CONSTEXPR_AFTER_CXX11_COMPLEX
+struct __abcd_results
+{
+    _Tp __ac;
+    _Tp __bd;
+    _Tp __ad;
+    _Tp __bc;
+};
+
+template<class _Tp>
+struct __ab_results
+{
+    _Tp __a;
+    _Tp __b;
+};
+
+template<class _Tp>
```

Comment:
I am not a big fan of SFINAE only on one of the functions. I would prefer if we could go with the more explicit
```suggestion
template<class _Tp, __enable_if_t<!__is_half_floating_point<_Tp>, int> = 0>
```

Reply:
...yeah that SFINAE was wrong, missing `__` before `type`. Will do an equivalent of this, sans enabling it for vector enabled types (which set may increase) and not just on half fp.